### ISSUE TITLE
Add a parallel version of foreach

### ISSUE THREAD

**Issue by drewhk** [akka/akka#17461]
Basically just sugar around mapAsyncUnordered. Looks like:

    source.foreachParallel(4) { doStuff() }

Can be useful for accepting connections from a connection source (TCP or HTTP)

**Comment by agolubev** [comment]
should it be something like 
```scala
def foreachParallel[T](parallelism: Int)(f: Out => T): Repr[T, Mat] =
    andThen(MapAsyncUnordered(
      parallelism,
      {out: Out => Future(f(out))}.asInstanceOf[Any ⇒ Future[Any]]))
```
for scala dsl?

**Comment by drewhk** [comment]
yes, exactly :) (+Java counterpart)

**Comment by agolubev** [comment]
Ok, let me make few statements here to be sure everything will be working as expected:
* ```foreachParallel``` will take function like  ```f:Out => Unit``` as an argument
* return type for ```foreachParallel``` will be ```Future[Unit]```
* when some ```f``` fails with exception or several ```f``` failed, ```foreachParallel``` will fail ```Future```  with first exception when stream is completed (if ```supervisionStrategy(resumingDecider)```) - this is what I'm not sure about
* when some ```f```` fails with exception or many exceptions, ```foreachParallel``` will fail Future at once with  exception  (if ```supervisionStrategy(stop)```)
* ```foreachParallel``` will be placed in class ```scaladsl.Source```
* as in ```Source``` there is method ```runForeach```, our final method name will be ```runForeachParallel``` (instead of foreachParallel)

Basically we can have four exception handling policies here:
* stop on each exception
* ignore all exception
* failed future on complete with first exception
* failed future on complete with last exception

So for three cases out of four most exceptions are lost. Still, maybe it would be good to drop exceptions with ```ForeachParallel``` in "optimistic mode". However, if it matters we can compose stream with ```recoverWith```
